src/ExtractImportGraph.elm

```diff
@@ -142,52 +169,63 @@ moduleVisitor schema =
         |> Rule.withImportVisitor importVisitor
 
 
-finalEvaluationForProject : ProjectContext -> List (Rule.Error { useErrorForModule : () })
-finalEvaluationForProject projectContext =
+dataExtractor : ProjectContext -> Encode.Value
+dataExtractor projectContext =
     let
-        toNode : List String -> String
-        toNode mN =
-            "\"" ++ String.join "." mN ++ "\""
-    in
-    [ Rule.globalError
-        { message = "Import Graph"
-        , details =
-            List.filterMap
-                (\( { name, isSourceModule }, imports ) ->
-                    let
-                        relevantImports =
-                            List.filter (not << .isDependency) imports
-                    in
-                    if List.isEmpty relevantImports || not isSourceModule then
-                        Nothing
-
-                    else
-                        toNode name
+        nodes : List String
+        nodes =
+            projectContext.imports
+                |> Dict.toList
+                |> List.map
+                    (\( name, imports ) ->
+                        "  "
+                            ++ wrapNameInQuotes name
                             ++ " -> {"
-                            ++ String.join " " (List.map (toNode << .name) relevantImports)
+                            ++ imports
                             ++ "}"
-                            |> Just
-                )
-                projectContext.imports
-                |> (\es -> "digraph {" :: es ++ [ "}" ])
-        }
-    ]
+                    )
+
+        graph : String
+        graph =
+            "digraph {\n" ++ String.join "\n" nodes ++ "\n}"
+    in
+    Encode.object
+        [ ( "onlineGraph", Encode.string ("https://dreampuf.github.io/GraphvizOnline/#" ++ escapeUrlCharacters graph) )
+        , ( "graph", Encode.string graph )
+        ]
+
+
+escapeUrlCharacters : String -> String
```

Comment:
Would it be better to use the core [Url.percentEncode](https://package.elm-lang.org/packages/elm/url/latest/Url#percentEncode)?  Or just trying to avoid the dependency?

Reply:
That sounds good :D